### ISSUE TITLE
Server logs aren't printing out any of the print statements and outputs a warning

### ISSUE THREAD

**Issue by valin1** [e-mission/e-mission-server#629]
I made some changes in the cfc_webapp.py, as first I changed the routing like what you mentioned yesterday to have different paths, but for some reason, it causes an error on the phone side. 
So, I thought that I would change the file paths to what they were originally suggestion_sys, since the main suggestion page and single suggestions were able to output something despite the single suggestions not having the correct suggestion. 

However, I changed it back to the routing paths on the server and here is what I have. (Note: the names of the file paths are different than what I have on the pull request)
```
@post('/suggestion_sys/getSug')
def getSuggestion():
  logging.debug("Called suggestion")
  user_uuid=getUUID(request)
  logging.debug("user_uuid %s" % user_uuid)
  ret_dir = suggsys.calculate_yelp_server_suggestion_nominatim(user_uuid)
  logging.debug("type(ret_dir) = %s" % type(ret_dir))
  logging.debug("Output of ret_dir = %s" % ret_dir)
  return ret_dir

@post('/suggestion_sys/getSing/<tripid>')
def getSingleTripSuggestion(tripid):
  logging.debug("Called suggestion.getSingleTrip"))
  user_uuid=getUUID(request)
  logging.debug("user_uuid %s" % user_uuid)
  ret_dir = suggsys.calculate_yelp_server_suggestion_singletrip_nominatim(user_uuid, tripid)
  logging.debug("type(ret_dir) = %s" % type(ret_dir))
  logging.debug("Output of ret_dir = %s" % ret_dir)
  return ret_dir
``` 

We keep getting as the server output: 
``` 
START 2019-01-25 04:52:05.662198 POST /suggestion_sys
END 2019-01-25 04:52:05.662981 POST /suggestion_sys  0.0005040168762207031
``` 

Along with the warning about Deprecation.

**Comment by valin1** [comment]
I am not exactly sure what do you mean by the phone code?

**Comment by valin1** [comment]
ohhhh do you mean sam's code?

**Comment by shankari** [comment]
yes, if you define an API path that like `/foo/bar` then the API client (the phone in this case) needs to call `/foo/bar` as well. If it calls `/foo` it is not going to be mapped to anything

**Comment by sam11b** [comment]
this.getSuggestion = function() {
      return new Promise(function(resolve, reject) {
        var msgFiller = function(mesage) {
          // donothing
        }
        window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys", msgFiller, resolve, reject);
      });
    };

    this.getSingleTripSuggestion = function(tripId) {
      return new Promise(function(resolve, reject) {
        var msgFiller = function(mesage) {
          // donothing
        }
        window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys", msgFiller, resolve, reject);
      });
    };

**Comment by sam11b** [comment]
^that is the services.js code for the two function calls

**Comment by sam11b** [comment]
$scope.getIndividualSuggestion = function() {
    $ionicLoading.show({
        template: 'Loading...'
        });
    CommHelper.getSingleTripSuggestion($stateParams.tripId).then(function(result) {
      console.log(result);
      $ionicLoading.hide();
      $scope.name = result.message;
      $scope.mode = result.method;
    }).catch(function(err) {
      console.log("Error while getting individual suggestion" + err);
    });
  };

**Comment by sam11b** [comment]
^this is the code for when the button to get an individual suggestion is pushed

**Comment by shankari** [comment]
This is directly related to https://github.com/e-mission/e-mission-server/issues/629#issuecomment-457464308
The server is now expecting either `/suggestion_sys/getSug` or `/suggestion_sys/getSing/<tripid>`
but you are calling 

```
    window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys", msgFiller, resolve, reject);
```

which is obviously going to fail

**Comment by sam11b** [comment]
now we are getting a com.google.GTMSessionFetcher error -1

I've tried looking at this error online and I can't seem to figure out what it means or how to approach it

**Comment by shankari** [comment]
what is the actual error?

**Comment by shankari** [comment]
can you copy and paste the error including a few lines before and after in the log?

**Comment by sam11b** [comment]
The logs are showing no error and are saying suggestion_sys is not being called.

**Comment by sam11b** [comment]
this.getSuggestion = function() {
      return new Promise(function(resolve, reject) {
        var msgFiller = function(mesage) {
          // donothing
        }
        window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys/getSug", msgFiller, resolve, reject);
      });
    };

    this.getSingleTripSuggestion = function(tripId) {
      return new Promise(function(resolve, reject) {
        var msgFiller = function(mesage) {
          // donothing
        }
        window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys/getSing/<tripId>", msgFiller, resolve, reject);
      });
    };

**Comment by valin1** [comment]
@shankari We aren't getting any error from the logs and suggestion_sys isn't getting called at all?

**Comment by sam11b** [comment]
^ I changed the pathways so I'm unsure why the system is not calling the functions

**Comment by shankari** [comment]
you are not getting any error in which logs?
there are two sets of logs - the server logs and the client logs
@sam11b just said
> now we are getting a com.google.GTMSessionFetcher error -1
that must show up in the client logs

So please copy/paste that error along with the lines before and after it

**Comment by sam11b** [comment]
<img width="1000" alt="screen shot 2019-01-24 at 10 40 37 pm" src="https://user-images.githubusercontent.com/23513222/51729545-1f286600-2029-11e9-96e9-aba69f8d592b.png">

**Comment by valin1** [comment]
We aren't get errors from the server logs.

**Comment by shankari** [comment]
Are any calls making it to the server? (e.g. diary display)?

**Comment by valin1** [comment]
@shankari  The main rec page is working now.

**Comment by valin1** [comment]
@shankari  but still not sure what's wrong with the each trip suggestion

**Comment by shankari** [comment]
so what was wrong with the main rec page? just curious?

**Comment by shankari** [comment]
wrt each trip suggestion, this is the code that @sam11b has

```
    window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys/getSing/<tripId>", msgFiller, resolve, reject);
```

how do you ensure that this will retrieve `tripid1` versus `tripid2`? Did you look at other examples of functions with variables passed in to them?

**Comment by sam11b** [comment]
$scope.getIndividualSuggestion = function() {
    $ionicLoading.show({
        template: 'Loading...'
        });
    CommHelper.getSingleTripSuggestion($stateParams.tripId).then(function(result) {
      console.log(result);
      $ionicLoading.hide();
      $scope.name = result.message;
      $scope.mode = result.method;
    }).catch(function(err) {
      console.log("Error while getting individual suggestion" + err);
    });
  };

**Comment by sam11b** [comment]
on the diary page i pass in the tripId for that specific trip. I tested it and it correctly uses the right tripID for the given trip on the page.

**Comment by sam11b** [comment]
This error is still occurring. The server said there was a syntax error so we addressed it but now that error is still coming up when I press the singleTrip button.

**Comment by shankari** [comment]
> on the diary page i pass in the tripId for that specific trip. I tested it and it correctly uses the right tripID for the given trip on the page.

How did you test it? Because what I see in the implementation of `getSingleTripSuggestion` is a hardcoded string (`tripId`) in the URL (`/suggestion_sys/getSing/<tripId>`)

**Comment by valin1** [comment]
> wrt each trip suggestion, this is the code that @sam11b has
> 
> ```
>     window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys/getSing/<tripId>", msgFiller, resolve, reject);
> ```
> how do you ensure that this will retrieve `tripid1` versus `tripid2`? Did you look at other examples of functions with variables passed in to them?

Are you asking about the webapp functions?

**Comment by shankari** [comment]
Again, please look at other functions that pass an argument to the function as part of the URL and see how they work.

**Comment by shankari** [comment]
> Are you asking about the webapp functions?

No, I talked about @sam11b's code

**Comment by shankari** [comment]
ok, so which is the other function call that passes in an argument as part of the URL?

**Comment by sam11b** [comment]
getMetrics is an example

**Comment by shankari** [comment]
ok, so what is the URL defined in the API call?

**Comment by sam11b** [comment]
Ah I see what you're saying. I'm changing the code I'll let you know if it works. So sorry!

**Comment by sam11b** [comment]
this.getSingleTripSuggestion = function(tripId) {
      return new Promise(function(resolve, reject) {
        var msgFiller = function(mesage) {
          // donothing
        }
        window.cordova.plugins.BEMServerComm.pushGetJSON("/suggestion_sys/getSing/" +tripId, msgFiller, resolve, reject);
      });
    };

**Comment by shankari** [comment]
@sam11b that looks a lot better

**Comment by sam11b** [comment]
Updated code accounting for my previous error (sorry!). Now I'm getting an HTTP status 500 error so there's something going on in the backend.

**Comment by shankari** [comment]
yup, 500 error is almost certainly something on the server (syntax error?)

**Comment by valin1** [comment]
@shankari We are having the server problem, because the input that I use for my function is an ObjectID(tripid) while Sam's tripid is a string, so when I try to query for entries with the same tripid that Sam has given me, my function doesn't work because the query requires a specific ObjectID type containing the tripid. We tried to see if there was anyway to obtain the same ObjectID type from the phone side, but it seems that all the ids are strings. I've also attempted to pass in tripids as strings while building my function, but strings never worked so I just kept with whatever object type the original id is. I was wondering if there is anyway to merge these ids together or fix this discrepancy of types?

**Comment by shankari** [comment]
@valin1 @sam11b good job debugging! now let's see if you can take the next step. There are effectively two inputs to your function, are there not? what is the input other than the `tripId`? how is it handled on the server?

**Comment by shankari** [comment]
> Is there someway to cast the String to an ObjectID?

Let's see if you can figure that out. Do you think that the uuid comes in as an `ObjectID` from the phone?

**Comment by valin1** [comment]
ohhhhhh, does it come in as a string?

**Comment by shankari** [comment]
well, you said earlier:

> We tried to see if there was anyway to obtain the same ObjectID type from the phone side, but it seems that all the ids are strings.

**Comment by valin1** [comment]
On the server side, it seems that it grabs the UUID from the token though? So it doesn't rely on the phone getting the UUID string? 

I am looking at the auth.py btw.

**Comment by shankari** [comment]
hm good point, we do convert the email from the token to the uuid. But there are other scripts where we pass in the uuid as a command line argument, right? For example the script to load data or to run the pipeline?

**Comment by valin1** [comment]
I am looking at the emission.core.wrapper.user.py and comparing it against trip.py because I'm assuming that file also creates a trip object and for the loading the data, they seem to register the UUID from the given useremail and creates an anonUUID. For the trip file, I am not exactly sure if I should be creating a new Trip instance after getting the tripid?

**Comment by shankari** [comment]
so `emission.core.wrapper.user` is a wrapper for the user object, so it expects to generally be passed in `ObjectId`s (kind of like your function). You need to find look at the scripts that take an a uuid through the CLI. scripts live in `bin` not in `emission`

**Comment by valin1** [comment]
I was looking at the load_timeline_for_day_and_user.py, so that's why I mentioned the emission.core.wrapper.user.py since it was imported as a module and they registered the user through that module. 
```
user = ecwu.User.register(args.user_email)
```

**Comment by shankari** [comment]
yes, but `load_timeline_for_day_and_user.py` doesn't take a uuid as an argument.
Think of the other scripts there...

```
$ ./e-mission-py.bash bin/debug/load_timeline_for_day_and_user.py
Connecting to database URL localhost
usage: load_timeline_for_day_and_user.py [-h] [-r] [-v VERBOSE]
                                         timeline_filename user_email
load_timeline_for_day_and_user.py: error: the following arguments are required: timeline_filename, user_email
```

**Comment by valin1** [comment]
ohhh wait it takes an email and then converts or registers it as a UUID though?

**Comment by shankari** [comment]
yes and the registration function already returns an `ObjectID` so it doesn't help you with your specific problem

**Comment by valin1** [comment]
i don't think the intake file for cleaning helps my problem either?

**Comment by shankari** [comment]
if this is taking too long, another way to tackle it is by .... looking at the documentation :)
so you want to convert a string to an `ObjectId`. which package does the `ObjectId` come from?

**Comment by valin1** [comment]
oh wait there is another option of uuid, ohhhhh so are you saying i should be using the uuid module to essentially turn that id into a uuid or i guess the object id?

**Comment by valin1** [comment]
sorry but with objectid?

**Comment by valin1** [comment]
ohhh okay i think i got it!

**Comment by shankari** [comment]
ok so post the solution here once you have figured it out?

**Comment by shankari** [comment]
or even post step by step what you are considering so that I can stop you before you go too far off track...

**Comment by valin1** [comment]
```
import bson
test_id_1 =bson.objectid.ObjectId('inserttripid')
``` 
I omitted the tripid but here is what it would look like. I tested it on ipython and my function works fine as usual.

**Comment by shankari** [comment]
yup! that's it! did you figure it out from the docs or the example?
I generally import the package with a shorter name to avoid redundancy

```
import bson.objectid as boi
test_id_1 = boi.ObjectId('inserttripid')
```

**Comment by valin1** [comment]
I figured it out from the docs.

**Comment by shankari** [comment]
@valin1 please reflect these changes in the PR so that I can merge it.

**Comment by shankari** [comment]
Fixed by #624
